model-optimizer/mo/front_ng/extractor.py

```diff
@@ -0,0 +1,159 @@
+    raise Error('There is no node with name {}'.format(node_name))
+
+
+def fe_input_user_data_repack(inputModel, input_user_shapes: [None, list, dict, np.ndarray],
+                              freeze_placeholder: dict, input_user_data_types=dict()):
+    """
+    Restructures user input cutting request. Splits ports out of node names.
+        Transforms node names to node ids.
+    :param graph: graph to operate on
+    :param input_user_shapes: data structure representing user input cutting request. It may be:
+    # None value if user did not provide neither --input nor --input_shape keys
+    # list instance which contains input layer names with or without ports if user provided
+        only --input key
+    # dict instance which contains input layer names with or without ports as keys and shapes as
+        values if user provided both --input and --input_shape
+    # np.ndarray if user provided only --input_shape key
+    :param freeze_placeholder: dictionary with placeholder names as keys and freezing value as values
+    :param input_user_data_types: dictionary with input nodes and its data types
+    :return: restructured input shapes and freeze placeholder shapes information
+    Example of input dictionary:
+    _input_shapes =
+    {
+        'node_ID':
+            [
+                {'shape': None, 'in': 0},
+                {'shape': None, 'in': 1},
+            ],
+        'node_1_ID':
+            [
+                {'shape': [1, 227, 227, 3], 'port': None, 'data_type': np.int32}
+            ],
+        'node_2_ID':
+            [
+                {'shape': None, 'out': 3}
+            ]
+    }
+     Example of freeze placeholder dictionary:
+    _freeze_placeholder =
+    {
+        'phase_train' : False
+    }
+    """
+    _input_shapes = []
+    # New version of FrontEnd is activated
+    print("Inside input_user_data_repack")
+    if isinstance(input_user_shapes, list) or isinstance(input_user_shapes, dict):
+        for input_name in input_user_shapes:
+            node = fe_decodeNameWithPort(inputModel, input_name)
+            if node is None:
+                raise Error('Cannot find location {} in the input model'.format(input_name))
+            shape = None if isinstance(input_user_shapes, list) else input_user_shapes[input_name]
+            if input_name in input_user_data_types and input_user_data_types[input_name] is not None:
+                data_type = input_user_data_types[input_name]
+                _input_shapes.append({'node': node, 'shape': shape, 'data_type': data_type})
+            else:
+                _input_shapes.append({'node': node, 'shape': shape})
+    elif isinstance(input_user_shapes, np.ndarray):
+        model_inputs = inputModel.get_inputs()
+        assert len(model_inputs) == 1
+        _input_shapes.append({'node': model_inputs[0], 'shape': input_user_shapes})
+    else:
+        assert input_user_shapes is None
+    # TODO: add logic for freeze_placeholder
```

Comment:
Ticket number or resolve the comment.

Reply:
@andrei-kochin Can you please create ticket to support 'freeze_placeholder' for frontends. Not sure that it should be targeted to 2022.1

Reply:
Ticket created